autolab/Autolab
# Add button to bulk import assessments from file system

## Summary

### Summary generated by Reviewpad on 16 Jul 23 19:01 UTC
This pull request includes several patches that make various changes to the codebase. Here is a summary of each patch:

1. [PATCH 1/7] Update old-style path to use path helper:
   - Makes a modification in the assessments_controller.rb file to update the old-style path to use the path helper. This change updates the redirect_to method call.

2. [PATCH 2/7] Add install all button:
   - Makes changes in the install_assessment.html.erb file to add an "Install all" button. The button triggers a script that sends a POST request for each import link and reloads the page after the import is done.

3. [PATCH 3/7] Rename install -> import:
   - Renames the "install" class to "import" in the install_assessment.html.erb file.

4. [PATCH 4/7] Improve logic:
   - Adds logic to display an alert message when the "Import all" button is clicked. The message indicates whether all assessments have been imported successfully, or if there were errors during the import process.

5. [PATCH 5/7] Add alert:
   - Adds an alert message to inform the user about the completion of the import process and any errors encountered during the import of assessments.

6. [PATCH 6/7] Rephrase message:
   - Modifies the alert message to rephrase the text when all assessments have been imported, but there are remaining assessments that encountered errors during the import process.

7. [PATCH 7/7] Rename btn variable:
   - Renames the 'btn' variable to '$btn' for clarity in the install_assessment.html.erb file.

Please review these patches and provide any necessary feedback or comments.


## Description

- Update an old-style path to use a path helper
- Improve UI of uninstalled assessments list
- Add button to bulk import assessments

## Motivation and Context



Currently, assessments can only be individually imported from the file system. This PR provides an option to import all of them at once.

Closes #1928

## How Has This Been Tested?



Create a course and a few assessments. Delete all of the assessments.
- Ensure that importing from file system still works
- Ensure that import all button works

<img width="1052" alt="Screenshot 2023-06-25 at 02 58 16" src="https://github.com/autolab/Autolab/assets/9074856/25a6667e-398c-491d-8cff-58884bc433f3">

## Types of changes


## Checklist:
